src/main/java/org/terning/user/vo/UserName.java

```diff
@@ -0,0 +1,62 @@
+package org.terning.user.vo;
+
+import jakarta.persistence.Embeddable;
+import lombok.EqualsAndHashCode;
+
+@Embeddable
+@EqualsAndHashCode
+public class UserName {
+
+    private static final int NAME_MAX_LENGTH = 12;
+    private static final String NAME_REGEX = "^[a-zA-Z0-9가-힣 ]*$";
+
+    private final String name;
+
+    protected UserName() {
+        this.name = null;
+    }
+
+    private UserName(String name) {
+        validateName(name);
+        this.name = name;
+    }
+
+    public static UserName from(String name) {
+        return new UserName(name);
+    }
+
+    public String value() {
+        return name;
+    }
+
+    // TODO: UserException 클래스를 만들어서 BaseException을 상속받아서 사용하도록 수정
+    // TODO: BaseException 구현 및 global 예외처리 구현
+    private void validateName(String name) {
+        validateNull(name);
+        validateLength(name);
+        validateInvalidCharacters(name);
+    }
+
+    private void validateNull(String name) {
+        if (name == null) {
+            throw new IllegalArgumentException("이름은 null일 수 없습니다.");
+        }
+    }
+
+    private void validateLength(String name) {
+        if (name.length() > NAME_MAX_LENGTH) {
+            throw new IllegalArgumentException("이름의 길이는 12자를 초과할 수 없습니다.");
```

Comment:
```suggestion
            throw new IllegalArgumentException("이름의 길이는 " + NAME_MAX_LENGTH + "자를 초과할 수 없습니다.");
```
보통 에러 던질 때는 매직넘버 처리를 사용하진 않나요? JUST 궁금

Reply:
피드백 감사합니다!

에러 메시지에서도 상수를 활용하면 유지보수에 유리하다는 점에 공감합니다!
현재는 커스텀 예외 처리나 메시지 관리 환경을 도입하기 전이라, 임시로 하드코딩된 메시지를 사용한 상태인데요.
말씀해주신 것처럼 `NAME_MAX_LENGTH` 상수를 에러 메시지에도 활용하면, 이후 상수 값이 변경되었을 때 메시지 누락 없이 반영되는 이점이 분명히 있는 것 같습니다.

예외 처리 구조를 정비하게 되면, 해당 피드백 참고해서 에러 메시지도 상수 기반으로 관리해보겠습니다. 

감사합니다! 🙏